pkg/workflow/hooks/data_passing.go

```diff
@@ -47,10 +47,10 @@ func Output(ctx wfContext.Context, taskValue *value.Value, step v1beta1.Workflow
 	if wfTypes.IsStepFinish(status.Phase, status.Reason) {
 		for _, output := range step.Outputs {
 			v, err := taskValue.LookupByScript(output.ValueFrom)
-			if err != nil {
+			if err != nil && !strings.Contains(err.Error(), "not found") {
```

Comment:
Would you extract it to be one util function, which decides the error is NotFound?

Reply:
This judgment is only for the outputs so I don't think we should extract it to a util function.